docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -640,6 +640,68 @@ The following example defines a CloudFormation resource that has the
         fooAlias: String
     }
 
+.. smithy-trait:: aws.cloudformation#cfnDefaultValue
+.. _aws.cloudformation#cfnDefaultValue-trait:
+
+----------------------------------------
+``aws.cloudformation#cfnDefaultValue`` trait
+----------------------------------------
+
+Summary
+    Indicates that the member annotated has a default value for the resource.
+Trait selector
+    ``structure > member``
+
+    *Any structure member*
+Value type
+    Annotation trait
+
+Given the following example, because the ``fooAlias``
+member is annotated with ``cfnDefaultValue``, it can be derived
+that the ``fooAlias`` member has a default value for this resource.
+
+.. code-block:: smithy
+
+    namespace smithy.example
+
+    use aws.cloudformation#cfnDefaultValue
+    use aws.cloudformation#cfnResource
+
+    @cfnResource
+    resource Foo {
+        identifiers: {
+            fooId: String,
+        },
+        read: GetFoo,
+    }
+
+    @readonly
+    @http(method: "GET", uri: "/foos/{fooId}", code: 200)
+    operation GetFoo {
+        input: GetFooRequest,
+        output: GetFooResponse,
+    }
+
+    @input
+    structure GetFooRequest {
+        @httpLabel
+        @required
+        fooId: String,
+
+        @cfnDefaultValue
```

Comment:
I wanted to ask about one more meaning for this trait: should it be applied to input or output values or both?

Based on how I understand the intent of this trait, it probably makes more sense to be attached to an output member shape associated with a property and not an input. Essentially if a describe operation occurs, and a client see's a value in a response field and they did not set the corresponding value on create or a prior update, they know they are seeing a defaulted. The refinement removes the ambiguity of a property likely being an input on both create and and update, and needing to specify `@cfnDefaultValue` on both. It's only relevant after calling Describe* and needing to assess what is going on regardless of what the last write/modify API called was correct?

This is something that can be refined in the selector.

Reply:
Refined the selector as the trait should be limited to scope in output operations only